### ISSUE TITLE
grunt: check for unused accessibility globals in course JSON

### ISSUE THREAD

**Issue by tomgreenfield** [adaptlearning/adapt_framework#2175]
Warn of potentially unneeded globals.

**Comment by oliverfoster** [comment]
Sounds like a plan.
1. Add missing attributes declared in schema (current behaviour)
2. Remove attributes undeclared in schema (new suggestion)

This would be a problem if code was compiled without an extension that was needed later but which had manually updated globals. The needed and amended attributes would be removed.
That's a rather slim use-case though.

**Comment by tomgreenfield** [comment]
I'd be slightly worried about an automated grunt task which removes unrecognised globals; was just pining for a human check of our JSON.

**Comment by lc-thomasberger** [comment]
I would only delete it if a flag is set by the user. Maybe the task should just flag potential unused globals.

**Comment by oliverfoster** [comment]
Original issue moved to #2200, this issue renamed to reflect requests.

**Comment by oliverfoster** [comment]
this issue is only relevent for hand-built courses and only when removing plugins, it's not work the effort, closing for now
